util/src/main/java/com/google/cloud/hadoop/util/EntriesCredentialConfiguration.java

```diff
@@ -256,6 +276,24 @@ public void setConfiguration(Entries entries) {
         setEnableServiceAccounts(enableServiceAccounts.get());
       }
 
+      // Parameters for SA directly in Configuration
+      String saPrivateKeyId = entries.getPassword(prefix + SA_PRIVATE_KEY_ID_SUFFIX);
+      if (saPrivateKeyId != null) {
+        setSaPrivateKeyId(saPrivateKeyId);
+      }
+
+      String saPrivateKey = entries.getPassword(prefix + SA_PRIVATE_KEY_SUFFIX);
+      if (saPrivateKey != null) {
+        saPrivateKey = saPrivateKey.replace("\\n", System.lineSeparator());
```

Comment:
Should this be move in the `setSaPrivateKey` method? if it should be done always.

Reply:
Sure.